File: content-solutions/dfs/radio-network.java
class Solution {
    private int[][] link;
    private boolean[] visited;
    private int n;

    int dfs(int v) {
        visited[v] = true;
        int size = 1;
        for (int w = 0; w < n; w++) {
            if (link[v][w] == 1 && !visited[w]) size += dfs(w);
        }
        return size;
    }

    public int[] solution(int[][] link) {
        this.link = link;
        n = link.length;
        visited = new boolean[n];
        int groups = 0, largest = 0;
        for (int v = 0; v < n; v++) {
            if (!visited[v]) {
                groups++;
                largest = Math.max(largest, dfs(v));
            }
        }
        return new int[] {groups, largest};
    }
}

File: content-solutions/dfs/radio-network.js
function solution(link) {
  const n = link.length;
  const visited = new Array(n).fill(false);
  function dfs(v) {
    visited[v] = true;
    let size = 1;
    for (let w = 0; w < n; w += 1) {
      if (link[v][w] === 1 && !visited[w]) size += dfs(w);
    }
    return size;
  }
  let groups = 0;
  let largest = 0;
  for (let v = 0; v < n; v += 1) {
    if (!visited[v]) {
      groups += 1;
      largest = Math.max(largest, dfs(v));
    }
  }
  return [groups, largest];
}
